scripts/prep-release.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# Helper script to prepare a release for the Go SDK.
+
+# Read the current build number from version-build to make sure the build number has been updated
+current_build_number=$(< internal/version-build)
+
+version_file="internal/version"
+build_file="internal/version-build"
+
+enforce_latest_code() {
+    if [[ -n "$(git status --porcelain=v1)" ]]; then
+        echo "ERROR: working directory is not clean."
+        echo "Please stash your changes and try again."
+        exit 1
+    fi
+}
+
+# Function to validate the version number format x.y.z(-beta.w)
+update_and_validate_version() {
+    while true; do
+        # Prompt the user to input the version number
+        read -p "Enter the version number (format: x.y.z(-beta.w)): " version
+
+        # Validate the version number format
+        if [[ "${version}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-beta\.[0-9]+)?$ ]]; then        
+            # Write the valid version number to the file
+            echo "${version}" > "${version_file}"
+            echo "New version number is: ${version}"
+            return 0
+        else
+            echo "Invalid version number format: ${version}"
+            echo "Please enter a version number in the 'x.y.z(-beta.w)' format."
+        fi
+    done
+}
+
+# Function to validate the build number format.
+# SEMVER Format: Mmmppbb - 7 Digits 
+update_and_validate_build() {
+    while true; do
+        # Prompt the user to input the build number
+        read -p "Enter the build number (format: Mmmppbb): " build
+
+        # Validate the build number format
+        if [[ "${build}" =~ ^[0-9]{7}$ ]]; then
```

Comment:
Nit: Feel free to skip for now, but we could also validate that the build number is not equal or less than the current build number.

Reply:
Done!